From 364b6bf5c142544b442c6d69911839edca9e994b Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Wed, 21 Sep 2016 01:18:41 +0200
Subject: Revert "wined3d: Implement the initial texture data upload on top of
 wined3d_device_update_sub_resource()."

This reverts commit e58689a3981fae5b312b12e9f6801adf289d04c0.
---
 dlls/wined3d/texture.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/dlls/wined3d/texture.c b/dlls/wined3d/texture.c
index c82016e5..dddd2fa 100644
--- a/dlls/wined3d/texture.c
+++ b/dlls/wined3d/texture.c
@@ -2828,6 +2828,7 @@ HRESULT CDECL wined3d_texture_create(struct wined3d_device *device, const struct
     if (data)
     {
         unsigned int sub_count = level_count * layer_count;
+        struct wined3d_context *context;
         unsigned int i;
 
         for (i = 0; i < sub_count; ++i)
@@ -2841,11 +2842,21 @@ HRESULT CDECL wined3d_texture_create(struct wined3d_device *device, const struct
             }
         }
 
+        context = context_acquire(device, NULL);
+
+        wined3d_texture_prepare_texture(object, context, FALSE);
+        wined3d_texture_bind_and_dirtify(object, context, FALSE);
+
         for (i = 0; i < sub_count; ++i)
         {
-            wined3d_device_update_sub_resource(device, &object->resource,
-                    i, NULL, data[i].data, data[i].row_pitch, data[i].slice_pitch);
+            const struct wined3d_const_bo_address addr = {0, data[i].data};
+
+            wined3d_texture_upload_data(object, i, context, &addr, data[i].row_pitch, data[i].slice_pitch);
+            wined3d_texture_validate_location(object, i, WINED3D_LOCATION_TEXTURE_RGB);
+            wined3d_texture_invalidate_location(object, i, ~WINED3D_LOCATION_TEXTURE_RGB);
         }
+
+        context_release(context);
     }
 
     TRACE("Created texture %p.\n", object);
-- 
2.9.0

